packages/blade/src/components/FileUpload/_decisions/decisions.md

```diff
@@ -0,0 +1,398 @@
+# FileUpload Decisions <!-- omit in toc -->
+
+The `FileUpload` component is used to handle file attachments, including the drag-and-drop interaction. It can be used in both controlled and uncontrolled forms. Primarily, it is used to upload files to a server or to display a list of uploaded files.
+
+This document outlines the API details of the `FileUpload` component, encompassing their structural composition, functional attributes, and visual representation.
+
+<img width="100%" src="./single-upload-variant.png" alt="Single File Upload">
+<img width="100%" src="./multiple-upload-variant.png" alt="Multiple Files Upload">
+<img width="100%" src="./file-upload-item-variants.png" alt="FileUploadItem Variants ">
+
+- [Design](#design)
+- [Anatomy](#anatomy)
+- [`FileUpload` Props](#fileupload-props)
+- [Usage](#usage)
+  - [Submit Button File Upload (Uncontrolled)](#submit-button-file-upload-uncontrolled)
+  - [Direct Upload with Progress Display (Controlled)](#direct-upload-with-progress-display-controlled)
+- [Examples](#examples)
+  - [Uncontrolled Usage](#uncontrolled-usage)
+    - [Single File selection](#single-file-selection)
+    - [Multiple File selection:](#multiple-file-selection)
+  - [Controlled Usage](#controlled-usage)
+    - [Direct Upload with Progress Display](#direct-upload-with-progress-display)
+- [Accessibility](#accessibility)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade-DSL?type=design&node-id=78150%3A2625&mode=design&t=bKQvPInRmAoeAAFV-1) to all variants of the `FileUpload` component.
+
+## Anatomy
+
+<img width="100%" src="./file-upload-anatomy.png" alt="Single File Upload">
+<img width="100%" src="./file-upload-item-anatomy.png" alt="Multiple Files Upload">
+
+## `FileUpload` Props
+
+```ts
+// Check the File type from MDN for more details: https://developer.mozilla.org/en-US/docs/Web/API/File
+interface BladeFile extends File {
+  /**
+   * The unique identifier of the file.
+   */
+  id: string;
+  /**
+   * The file's upload status.
+   */
+  status?: 'uploading' | 'success' | 'error';
+  /**
+   * The percentage of file upload completion.
+   */
+  percent?: number;
```

Comment:
Instead of `percent` should we use `progressValue`, `progressMin`, & `progressMax`? You will need to pass these to Progress bar component anyway and the way we accept "progress" related props can be similar here as well as in Progress bar

Reply:
I don't think we need `progressMin` & `progressMax` it would always be 0 & 100 for a file being uploaded. We can rename `percent` to `uploadPercent` maybe.

Reply:
Yeah. Either call it `uploadPercent` or `ProgressPercent`

Reply:
Updated